libs/features/cdt/cdt.js

```diff
@@ -0,0 +1,123 @@
+import { getMetadata, getConfig, createTag } from '../../utils/utils.js';
+import { replaceKey } from '../placeholders.js';
+
+function loadCountdownTimer(
+  container,
+  cdtLabel,
+  cdtDays,
+  cdtHours,
+  cdtMins,
+  timeRangesEpoch,
+) {
+  let isVisible = false;
+  let interval;
+
+  function appendTimerBox(parent, value, label) {
+    const fragment = createTag('div', { class: 'timer-fragment' }, null, { parent });
+    const unitContainer = createTag('div', { class: 'timer-unit-container' }, null, { parent: fragment });
+    createTag('div', { class: 'timer-unit-label' }, label, { parent: fragment });
+
+    createTag('div', { class: 'timer-box' }, Math.floor(value / 10).toString(), { parent: unitContainer });
+    createTag('div', { class: 'timer-box' }, (value % 10).toString(), { parent: unitContainer });
+  }
+
+  function appendSeparator(parent) {
+    createTag('div', { class: 'timer-separator' }, ':', { parent });
+  }
+
+  function appendTimerBlock(parent, daysLeft, hoursLeft, minutesLeft) {
+    const timerBlock = createTag('div', { class: 'timer-block' }, null, { parent });
+    appendTimerBox(timerBlock, daysLeft, cdtDays);
+    appendSeparator(timerBlock);
+    appendTimerBox(timerBlock, hoursLeft, cdtHours);
+    appendSeparator(timerBlock);
+    appendTimerBox(timerBlock, minutesLeft, cdtMins);
+  }
+
+  function appendTimerLabel(parent, label) {
+    createTag('div', { class: 'timer-label' }, label, { parent });
+  }
+
+  function removeCountdown() {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function render(daysLeft, hoursLeft, minutesLeft) {
+    if (!isVisible) return;
+
+    removeCountdown();
+
+    appendTimerLabel(container, cdtLabel);
+    appendTimerBlock(container, daysLeft, hoursLeft, minutesLeft);
+  }
+
+  function updateCountdown() {
+    const instant = new URL(window.location.href)?.searchParams?.get('instant');
+    const currentTime = instant ? new Date(instant) : Date.now();
+
+    for (let i = 0; i < timeRangesEpoch.length; i += 2) {
+      const startTime = timeRangesEpoch[i];
+      const endTime = timeRangesEpoch[i + 1];
+
+      if (currentTime >= startTime && currentTime <= endTime) {
+        isVisible = true;
+        const diffTime = endTime - currentTime;
+        const daysLeft = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+        const hoursLeft = Math.floor((diffTime % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        const minutesLeft = Math.floor((diffTime % (1000 * 60 * 60)) / (1000 * 60));
+        render(daysLeft, hoursLeft, minutesLeft);
+        return;
+      }
+    }
+
+    isVisible = false;
+    clearInterval(interval);
+    removeCountdown();
+  }
+
+  function startCountdown() {
+    const oneMinuteinMs = 60000;
+    updateCountdown();
+    interval = setInterval(updateCountdown, oneMinuteinMs);
+  }
+
+  startCountdown();
+}
+
+function isMobile() {
+  return window.matchMedia('(max-width: 767px)').matches;
```

Comment:
The value gets cached and will not work when rotating screens. You need to refactor this to always see if the viewport matches the width. 
Example:
```js
const isMobile = window.matchMedia('(max-width: 767px)');
if (isMobile.matches) doSomething(); // this will work after you resize or rotate screens.
```